tracing/setup.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/tracing/config"
+)
+
+// SetupStaticPublishing sets up trace publishing for the process. Note that other
+// pieces still need to generate the traces, this just ensures that if generated, they are collected
+// appropriately. This is normally done by using tracing.HTTPSpanMiddleware as a middleware HTTP
+// handler. The configuration will not be dynamically updated.
+func SetupStaticPublishing(logger *zap.SugaredLogger, serviceName string, cfg *config.Config) error {
+	oct := NewOpenCensusTracer(WithExporter(serviceName, logger))
+	if err := oct.ApplyConfig(cfg); err != nil {
+		return fmt.Errorf("unable to set OpenCensusTracing config: %w", err)
+	}
+	return nil
+}
+
+// SetupDynamicPublishing sets up trace publishing for the process, by watching a
+// ConfigMap for the configuration. Note that other pieces still need to generate the traces, this
+// just ensures that if generated, they are collected appropriately. This is normally done by using
+// tracing.HTTPSpanMiddleware as a middleware HTTP handler. The configuration will be dynamically
+// updated when the ConfigMap is updated.
+func SetupDynamicPublishing(logger *zap.SugaredLogger, configMapWatcher *configmap.InformedWatcher, serviceName, tracingConfigName string) error {
```

Comment:
Final nit: this function never returns an error inside.Any reason it's in the interface?

Reply:
TBH I don't know :man_shrugging: , maybe it was returning an error in the past... Should I change the signature or we keep as is (maybe useful for forward compatibility, given at some point we'll switch to the new tracing lib)

Reply:
:shrug:
we can always change that.
I'd rather not have to check that we return an error when we shouldn't.
But it's fine. I can clean this up later, since it probably requires downstream changes.